plugins/inputs/huawei_routers_telemetry/huawei_routers_telemetry.go

```diff
@@ -0,0 +1,230 @@
+		if err != nil {
+			psl.Log.Errorf("Unable to decode incoming packet: %s", err.Error())
+		}
+		// Inicia el manejo de la telemetria
+		grouper := HuaweiTelemetryDecoder(body)
+		for _, metric := range grouper.Metrics() {
+			psl.AddMetric(metric)
+		}
+
+		if err != nil {
+			psl.Log.Errorf("Unable to parse incoming packet: %s", err.Error())
+
+			continue
+		}
+	}
+}
+
+type HuaweiRoutersTelemetry struct {
+	ServicePort     string        `toml:"service_port"`
+	ReadBufferSize  internal.Size `toml:"read_buffer_size"`
+	ContentEncoding string        `toml:"content_encoding"`
+	wg              sync.WaitGroup
+
+	Log telegraf.Logger
+
+	telegraf.Accumulator
+	io.Closer
+	decoder internal.ContentDecoder
+}
+
+func (sl *HuaweiRoutersTelemetry) Description() string {
+	return "Huawei Telemetry UDP model input Plugin"
+}
+
+func (sl *HuaweiRoutersTelemetry) SampleConfig() string {
+	return `
+  ## UDP Service Port to capture Telemetry
+  service_port = "8080"
+
+`
+}
+
+func (sl *HuaweiRoutersTelemetry) Gather(_ telegraf.Accumulator) error {
+	return nil
+}
+
+func (hrt *HuaweiRoutersTelemetry) Start(acc telegraf.Accumulator) error {
+	hrt.Accumulator = acc
+
+	var err error
+	hrt.decoder, err = internal.NewContentDecoder(hrt.ContentEncoding)
+	if err != nil {
+		return err
+	}
+
+	pc, err := udpListen("udp", ":"+hrt.ServicePort)
+	if err != nil {
+		return err
+	}
+
+	if hrt.ReadBufferSize.Size > 0 {
+		if srb, ok := pc.(setReadBufferer); ok {
+			srb.SetReadBuffer(int(hrt.ReadBufferSize.Size))
+		} else {
+			hrt.Log.Warnf("Unable to set read buffer on a %s socket", "udp")
+		}
+	}
+
+	hrt.Log.Infof("Listening on %s://%s", "udp", pc.LocalAddr())
+
+	psl := &packetSocketListener{
+		PacketConn:             pc,
+		HuaweiRoutersTelemetry: hrt,
+	}
+
+	hrt.Closer = psl
+	hrt.wg = sync.WaitGroup{}
+	hrt.wg.Add(1)
+	go func() {
+		defer hrt.wg.Done()
+		psl.listen()
+	}()
+	return nil
+}
+
+func udpListen(network string, address string) (net.PacketConn, error) {
+	switch network {
+	case "udp", "udp4", "udp6":
+		var addr *net.UDPAddr
+		var err error
+		var ifi *net.Interface
+		if spl := strings.SplitN(address, "%", 2); len(spl) == 2 {
+			address = spl[0]
+			ifi, err = net.InterfaceByName(spl[1])
+			if err != nil {
+				return nil, err
+			}
+		}
+		addr, err = net.ResolveUDPAddr(network, address)
+		if err != nil {
+			return nil, err
+		}
+		if addr.IP.IsMulticast() {
+			return net.ListenMulticastUDP(network, ifi, addr)
+		}
+		return net.ListenUDP(network, addr)
+	}
+	return net.ListenPacket(network, address)
+}
+
+func (hrt *HuaweiRoutersTelemetry) Stop() {
+	if hrt.Closer != nil {
+		hrt.Close()
+		hrt.Closer = nil
+	}
+	hrt.wg.Wait()
+}
+
+func newHuaweiRoutersTelemetry() *HuaweiRoutersTelemetry {
+	return &HuaweiRoutersTelemetry{}
+}
+
+type unixCloser struct {
+	path   string
+	closer io.Closer
+}
+
+func (uc unixCloser) Close() error {
+	err := uc.closer.Close()
+	os.Remove(uc.path) // ignore error
+	return err
+}
+
```

Comment:
I think this is not used.

Reply:
removed, I dont remeber why i need it. Sure in my initial testing